src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/transport/action/updateindexmetadata/UpdateManagedIndexMetaDataRequest.kt

```diff
@@ -24,38 +24,56 @@ import org.elasticsearch.common.io.stream.StreamOutput
 import org.elasticsearch.index.Index
 
 class UpdateManagedIndexMetaDataRequest : AcknowledgedRequest<UpdateManagedIndexMetaDataRequest> {
-    lateinit var listOfIndexMetadata: List<Pair<Index, ManagedIndexMetaData>>
+
+    lateinit var indicesToAddManagedIndexMetaDataTo: List<Pair<Index, ManagedIndexMetaData>>
+        private set
+
+    lateinit var indicesToRemoveManagedIndexMetaDataFrom: List<Index>
         private set
 
     constructor()
 
-    constructor(listOfIndexMetadata: List<Pair<Index, ManagedIndexMetaData>>) {
-        this.listOfIndexMetadata = listOfIndexMetadata
+    constructor(
+        indicesToAddManagedIndexMetaDataTo: List<Pair<Index, ManagedIndexMetaData>> = listOf(),
+        indicesToRemoveManagedIndexMetaDataFrom: List<Index> = listOf()
+    ) {
+        this.indicesToAddManagedIndexMetaDataTo = indicesToAddManagedIndexMetaDataTo
+        this.indicesToRemoveManagedIndexMetaDataFrom = indicesToRemoveManagedIndexMetaDataFrom
     }
 
     override fun validate(): ActionRequestValidationException? {
         var validationException: ActionRequestValidationException? = null
-        if (!this::listOfIndexMetadata.isInitialized) {
```

Comment:
Do we want && or || here?

Reply:
The constructor with parameters for ```UpdateManagedIndexMetaDataRequest``` will ensure that the lists are at least set to ```emptyList()``` which is how we invoke the request in our plugin. But if the values are being read in using ```readFrom()```, the lists could be null since the Transport class ```TransportUpdateManagedIndexMetaDataAction``` passes in the empty constructor as the ```Supplier```. Within that Transport class, I was referencing both lists (addTo/removeFrom) as if they were at most empty and not null so the validation was to confirm that assumption in the ```readFrom()``` case.

Reply:
After further discussion, we decided to use || to ensure both lists are initialized.